### PR TITLE
[DOCS] Connectors 8.16.0 release notes

### PR DESCRIPTION
**Pull request by leemthompo**
New connectors release notes now live in Elasticsearch guide

### CODE REVIEW

File: docs/reference/connector/docs/release-notes/connectors-release-notes-8.16.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
+[[es-connectors-release-notes-8-16-0]]
+=== 8.16.0 connectors release notes
+
+[discrete]
+[[es-connectors-release-notes-8-16-0-breaking]]
+==== Breaking Changes
+
+* Directly accessing connector and sync job state through `.elastic-connectors*` indices is deprecated, and will be disallowed entirely in a future release. Instead, the Elasticsearch Connector APIs should be used. Connectors framework code now uses the Elasticsearch Connector APIs by default.
+See https://github.com/elastic/connectors/pull/2884[*PR 2902*]..
```
Comment (artem-shelkovnikov) [inline docs/reference/connector/docs/release-notes/connectors-release-notes-8.16.0.asciidoc:9]: I'm not sure about this one - from what I understand we haven't fully migrated to Connector APIs and some endpoints might be missing cc @jedrazb.

Should we not be too strong with our deprecation statement here?

Comment (seanstory) [reply docs/reference/connector/docs/release-notes/connectors-release-notes-8.16.0.asciidoc:9]: No, we should be strong with the deprecation notice. It is going away _eventually_ and that's what we're trying to communicate. Not that it _cannot_ be used in 8.16 (so @leemthompo IDK if this belongs under "Breaking Changes."), but that, starting now, customers need to start planning to move away from direct index access.

File: docs/reference/connector/docs/release-notes/connectors-release-notes-8.16.0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
++
+Starting in `8.16.0`, two of our published Docker images and release artifacts now _also_ live under a new namespace: `integrations`, in addition to the existing `enterprise-search` namespace.
+This concerns the https://github.com/elastic/connectors[Elastic Connectors] and the https://github.com/elastic/data-extraction-service[Elastic Data Extraction Service].
++
+Please note that we recommend using the new `integrations` namespace, as we are consolidating integration Docker images there. We encourage users to adopt the new namespace as soon as possible.
++
+** *Example*:
++
+Old namespace::
+`docker.elastic.co/enterprise-search/elastic-connectors:v8.16.0`
```
Comment (seanstory) [inline docs/reference/connector/docs/release-notes/connectors-release-notes-8.16.0.asciidoc:25]: We should similarly note that the enterprise-search namespace is deprecated, and will eventually not house new connector releases, while the integrations space will.

**Review by kosabogi — APPROVED** [review]

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/116648)|
|✅|8.16|[](https://github.com/elastic/elasticsearch/pull/116649)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**